File: submissions/cpp/704.cpp
class Solution {
public:
    int search(vector<int> &nums, int target) {
        int i = 0, j = nums.size() - 1, index;
        while (i <= j) {
            int index = (i + j) / 2;
            if (nums[index] > target)
                j = index - 1;
            else if (nums[index] < target)
                i = index + 1;
            else
                return index;
        }
        return -1;
    }
};

File: submissions/Python3/704.py
class Solution:
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        start, end = 0, len(nums)- 1
        while start <= end:
            mid = start + (end - start) // 2
            if nums[mid] < target:
                start = mid + 1
            elif nums[mid] > target:
                end = mid - 1
            else:
                return mid
        return -1